### PR TITLE
Improve test

### PR DESCRIPTION
**Pull request by Morteza-Jenabzadeh**
hi
I'm interested in your package. I thought the test can be more accurate. 
when we check the value of `renderCount` in the Counter2 component we guarantee that Component2 will not be rerendered because of the main store changing.  And I think we don't need to button handler in the Counter2 component in this component we just want to check the rendering.

Best regards

### CODE REVIEW

**Comment by dai-shi** [comment]
Hi, thanks for your interest and suggestion!
Yes, improving tests would be nice, especially because v1.2 is more complex than v1.1 and we are preparing v2.
(Just a note: `renderCount` is testing internal implementation, for testing more accurately from outside, we should check commit count.)

File: __tests__/01_basic_spec.js [hunk 1]
```diff
@@ -69,6 +66,7 @@ describe('basic spec', () => {
     const { getAllByText, container } = render(<App />);
     expect(container).toMatchSnapshot();
     fireEvent.click(getAllByText('+1')[0]);
```
Comment (dai-shi) [inline __tests__/01_basic_spec.js:68]: Now, we have only one button, can we do this?
```suggestion
    fireEvent.click(getByText('+1'));
```

Comment (Morteza-Jenabzadeh) [reply __tests__/01_basic_spec.js]: how about this : 
```js
- fireEvent.click(getAllByText('+1')[0]);
+ fireEvent.click(screen.getByRole('button'));
```
it's more clear and easy to maintain(now it's applied in my PR)

and about what said 

>  for testing more accurately from outside, we should check commit count.


I will be happy to work on it if you give me a hint.

Comment (dai-shi) [reply __tests__/01_basic_spec.js]: By commit count, I mean we count with `useEffect`. You don't need to change the current test, but if you are willing to add more tests, that would be a good way.

**Review by dai-shi — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**